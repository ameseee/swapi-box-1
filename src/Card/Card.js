import React from 'react';
import PropTypes from 'prop-types';
import faveIcon from '../assets/favorite.png';
import notFaveIcon from '../assets/not-favorite.png';

const Card = ({ title, line1, line2, line3, line4, favoriteFunc, className }) => {
  const icon = (className === 'favorite') ? faveIcon : notFaveIcon;
  return (
    <div className={`card ${className}`}
      onClick={() => { favoriteFunc({ title, line1, line2, line3, line4 }); }}>
      <article>
        <img src={icon} alt='Text indicating if fave' />
        <h3>{title}</h3>
        <p>{line1}</p>
        <p>{line2}</p>
        <p>{line3}</p>
        <p>{line4}</p>
      </article>
    </div>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  line1: PropTypes.string,
  line2: PropTypes.string,
  line3: PropTypes.string,
  line4: PropTypes.string,
  favoriteFunc: PropTypes.func
};

export default Card;
