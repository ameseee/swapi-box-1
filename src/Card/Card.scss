.card {
  background-color: rgba(85, 140, 140, .5);
  width: 20%;
  margin: 25px;
  position: relative;
  text-align: left;
  border-radius: 5px;
  color: rgba(219, 219, 219, .85);
  -webkit-transition: all 200ms ease-in;
  -webkit-transform: scale(1);
  -ms-transition: all 200ms ease-in;
  -ms-transform: scale(1);
  -moz-transition: all 200ms ease-in;
  -moz-transform: scale(1);
  transition: all 200ms ease-in;
  transform: scale(1);
}

.card:hover {
  background-color: rgba(85, 140, 140, 1);
  color: rgba(0, 0, 0, 1);
  z-index: 2;
  -webkit-transition: all 200ms ease-in;
  -webkit-transform: scale(1.5);
  -ms-transition: all 200ms ease-in;
  -ms-transform: scale(1.5);
  -moz-transition: all 200ms ease-in;
  -moz-transform: scale(1.5);
  transition: all 200ms ease-in;
  transform: scale(1.1);
}

.card img {
  position: absolute;
  top: 0;
  right: 0;
}

.unfavorite img {
  width: 30px;
  padding: 4px;
}

.favorite img {
  width: 40px;
}

.favorite {
  transform: scale(1.1);
  background-color: rgba(85, 140, 140, 1);
  color: rgba(0, 0, 0, 1);
}

.card-cat {
  position: absolute;
  top: 0;
  left: 0;
  margin: 5px;
  font-size: .7em;
}

h3 {
  font-size: 1.4em;
  padding-top: 25px;
  margin-left: 5px;
}

p {
  margin-left: 5px;
}
