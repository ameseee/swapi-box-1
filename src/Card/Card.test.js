import React from 'react';
import Card from './Card';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('Card', () => {
  Enzyme.configure({ adapter: new Adapter() });
  let mockFn = jest.fn();
  let wrapper = shallow(<Card title='Leia Organa'
    line1='Homeworld: Alderaan'
    line2='Homeworld Population: 2000000000'
    line3='Species: Human'
    line4=''
    favoriteFunc={mockFn}
    className='favorite'/>);

  it('should render article', () => {
    const article = wrapper.find('article');

    expect(article.length).toEqual(1);
  });

  it('should render img according to className', () => {
    let favorite = wrapper.find('.favorite');
    let img = wrapper.find('img');
    // const src = img.find('src')

    expect(favorite.length).toEqual(1);
    expect(img.html()).toContain('favorite.png');

    const altWrapper = shallow(<Card title='Leia Organa'
      line1='Homeworld: Alderaan'
      line2='Homeworld Population: 2000000000'
      line3='Species: Human'
      line4=''
      favoriteFunc={mockFn}
      className='unfavorite'/>);
    favorite = altWrapper.find('.favorite');
    const unfavorite = altWrapper.find('.unfavorite');
    img = altWrapper.find('img');
    expect(favorite.length).toEqual(0);
    expect(unfavorite.length).toEqual(1);
    expect(img.html()).toContain('not-favorite.png');
  });

  it('should render title', () => {
    const title = wrapper.find('h3');

    expect(title.length).toEqual(1);
    expect(title.text()).toEqual('Leia Organa');
  });

  it('should render 4 <p> with corresponding text', () => {
    const lines = wrapper.find('p');
    const first = lines.first();
    const last = lines.last();
    const second = wrapper.findWhere(lines =>
      lines.text() === 'Homeworld Population: 2000000000');
    const third = wrapper.findWhere(lines => lines.text() === 'Species: Human');


    expect(lines.length).toEqual(4);
    expect(first.text()).toEqual('Homeworld: Alderaan');
    expect(last.text()).toEqual('');
    expect(second).toBeDefined();
    expect(third).toBeDefined();
  });

  it('should fire function when clicked', () => {
    const div = wrapper.find('div');

    div.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
