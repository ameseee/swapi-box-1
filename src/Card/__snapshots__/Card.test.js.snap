// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    className="favorite"
    favoriteFunc={[Function]}
    line1="Homeworld: Alderaan"
    line2="Homeworld Population: 2000000000"
    line3="Species: Human"
    line4=""
    title="Leia Organa"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <article>
        <img
                alt="Text indicating if fave"
                src="favorite.png"
        />
        <h3>
                Leia Organa
        </h3>
        <p>
                Homeworld: Alderaan
        </p>
        <p>
                Homeworld Population: 2000000000
        </p>
        <p>
                Species: Human
        </p>
        <p />
</article>,
      "className": "card favorite",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Text indicating if fave"
            src="favorite.png"
/>,
          <h3>
            Leia Organa
</h3>,
          <p>
            Homeworld: Alderaan
</p>,
          <p>
            Homeworld Population: 2000000000
</p>,
          <p>
            Species: Human
</p>,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "Text indicating if fave",
            "src": "favorite.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Leia Organa",
          },
          "ref": null,
          "rendered": "Leia Organa",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Homeworld: Alderaan",
          },
          "ref": null,
          "rendered": "Homeworld: Alderaan",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Homeworld Population: 2000000000",
          },
          "ref": null,
          "rendered": "Homeworld Population: 2000000000",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Species: Human",
          },
          "ref": null,
          "rendered": "Species: Human",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "article",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <article>
          <img
                    alt="Text indicating if fave"
                    src="favorite.png"
          />
          <h3>
                    Leia Organa
          </h3>
          <p>
                    Homeworld: Alderaan
          </p>
          <p>
                    Homeworld Population: 2000000000
          </p>
          <p>
                    Species: Human
          </p>
          <p />
</article>,
        "className": "card favorite",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Text indicating if fave"
              src="favorite.png"
/>,
            <h3>
              Leia Organa
</h3>,
            <p>
              Homeworld: Alderaan
</p>,
            <p>
              Homeworld Population: 2000000000
</p>,
            <p>
              Species: Human
</p>,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "Text indicating if fave",
              "src": "favorite.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Leia Organa",
            },
            "ref": null,
            "rendered": "Leia Organa",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Homeworld: Alderaan",
            },
            "ref": null,
            "rendered": "Homeworld: Alderaan",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Homeworld Population: 2000000000",
            },
            "ref": null,
            "rendered": "Homeworld Population: 2000000000",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Species: Human",
            },
            "ref": null,
            "rendered": "Species: Human",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "article",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
